signature/types.go

```diff
@@ -145,3 +145,21 @@ func (signerInfo *SignerInfo) ExtendedAttribute(key string) (Attribute, error) {
 	}
 	return Attribute{}, errors.New("key not in ExtendedAttributes")
 }
+
+// GetAuthenticSigningTime returns the authentic signing time
+func (signerInfo *SignerInfo) GetAuthenticSigningTime() (time.Time, error) {
+	switch signerInfo.SignedAttributes.SigningScheme {
+	case SigningSchemeX509SigningAuthority:
+		return signerInfo.SignedAttributes.SigningTime, nil
+	case SigningSchemeX509:
+		if len(signerInfo.UnsignedAttributes.TimestampSignature) > 0 {
+			// TODO: Add TSA support for AutheticSigningTime
+			// https://github.com/notaryproject/notation-core-go/issues/38
+			return time.Time{}, nil
```

Comment:
I might have miss guided you, We should return an error for this use case so that logic becomes simple.

if authentic singing time present return `time, nil` else return `time.Time{}, errros.new(
"")`